src/python/pants/util/strutil.py

```diff
@@ -307,3 +309,15 @@ def help_text(val: str | Callable[[], str]) -> str | Callable[[], str]:
         return softwrap(val)
     else:
         return lambda: softwrap(val())  # type: ignore[operator]
+
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+
+def docstring(doc: str | Callable[[], str]) -> Callable[[Callable[P, R]], Callable[P, R]]:
```

Comment:
Could you add a ... docstring ... to this function to explain when it should be used?